opencti-platform/opencti-graphql/tests/02-integration/04-manager/taskManager-test.ts

```diff
@@ -0,0 +1,96 @@
+import { afterAll, describe, expect, it } from 'vitest';
+import { executeReplace } from '../../../src/manager/taskManager';
+import type { AuthContext } from '../../../src/types/user';
+import { ADMIN_USER } from '../../utils/testQuery';
+import { MARKING_TLP_AMBER, MARKING_TLP_CLEAR } from '../../../src/schema/identifier';
+import { addReport, findById as findReportById } from '../../../src/domain/report';
+import { findById } from '../../../src/domain/markingDefinition';
+import { stixDomainObjectDelete } from '../../../src/domain/stixDomainObject';
+
+describe('TaskManager executeReplace tests ', () => {
+  const adminContext: AuthContext = { user: ADMIN_USER, tracing: undefined, source: 'taskManager-integration-test', otp_mandatory: false };
+  const reportsId: string[] = [];
+  afterAll(async () => {
+    expect(reportsId.length).toBe(3);
+    for (let index = 0; index < reportsId.length; index += 1) {
+      await stixDomainObjectDelete(adminContext, adminContext.user, reportsId[index]);
+      const report = await findReportById(adminContext, adminContext.user, reportsId[index]);
+      expect(report).toBeUndefined();
+    }
+  });
+  it('REPLACE report marking with different marking', async () => {
+    const reportInput = {
+      name: 'test report marking with different marking',
+      objectMarking: [MARKING_TLP_CLEAR],
+    };
+    const report = await addReport(adminContext, adminContext.user, reportInput);
+    expect(report.id).toBeDefined();
+    reportsId.push(report.id);
+    const reportId = report.id;
+    const actionContext = {
+      field: 'object-marking',
+      type: 'RELATION',
+      values: [MARKING_TLP_AMBER]
```

Comment:
RCA on the counter issues: this id here is supposed to be an internal id and not a standard id. In the task manager the new check compares to an internal id.

Reply:
fixed